python-39: update to 3.9.2 and switch to openssl-1.1

AndWac opened the pull request.

AndWac committed:
python-39: update to 3.9.2 and switch to openssl-1.1

alarcher reviewed (changes requested):
I am curious why all these tests fail now.

Also you should keep 21-sched_get_priority.patch as it is.
The version you reverted to is not entirely correct.

alarcher commented:
@AndWac Did you make sure that all the patches you removed were integrated to 3.9.2? It seems that tests regressed a lot from 3.9.1.

AndWac commented:
No, I will do that. I have checked mostly against solaris-userland and partly against OmniOS.

alarcher commented:
@AndWac We have more patches then they do, specifically to address test failures and fix asyncio. I added all the extra patches.

AndWac commented:
Am 08.03.21 um 16:05 schrieb Aurélien Larcher:
>
> @AndWac <https://github.com/AndWac> We have more patches then they do,
> specifically to address test failures and fix asyncio. I added all the
> extra patches.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/pull/6559#issuecomment-792815923>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADET4N2XSJGBVND7HAO73SLTCTRTFANCNFSM4YYAOLTQ>.
>
Sorry, I didn't know that. I will work on the PR either this evening or
tomorrow and will check everything again.
BTW on my systems I have typically more test failures than others. Some
of them probably due to my German locale settings.

AndWac commented:
Am 08.03.21 um 15:52 schrieb Aurélien Larcher:
>
> @AndWac <https://github.com/AndWac> Did you make sure that all the
> patches you removed were integrated to 3.9.2? It seems that tests
> regressed a lot from 3.9.1.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/pull/6559#issuecomment-792807365>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADET4N4H66JLC2PXIPJ6PX3TCTQCLANCNFSM4YYAOLTQ>.
>
Just for a data point:
I have run the tests on the actual python-3.9.1 with one additional
patch that skipped the readline tests.
The results on my machine were:

-405 tests OK.
+398 tests OK.

-2 tests failed:
-    test_epoll test_ssl
+9 tests failed:
+    test_argparse test_dtrace test_epoll test_httpservers test_import
+    test_shutil test_spwd test_ssl test_support

  18 tests skipped:
-    test_gdb test_kqueue test_msilib test_ossaudiodev test_smtpnet
+    test_gdb test_kqueue test_msilib test_readline test_smtpnet
      test_socketserver test_startfile test_timeout test_tix test_tk
      test_ttk_guionly test_urllib2net test_urllibnet test_winconsoleio
      test_winreg test_winsound test_xmlrpc_net test_zipfile64
gmake: *** [/oibuild/oi-userland/make-rules/configure.mk:222:
/oibuild/oi-userland/components/python/python39/build/amd64/.tested-and-compared]
Error 2

AndWac commented:
I have checked and fixed some things:
13-SOABI.patch.3.9.1                    seems to be obsolete / does not work
20-test_fixes.patch.3.9.1               no differences in test results
23-sendfile-reenable.patch.3.9.1        patches only differ in version
27-cgiserver.patch.3.9.1                patches only differ in version
28-gid-convert.patch.3.9.1              already in 3.9.2
31-regrtest-processgroup.patch.3.9.1    important: let 3 tests pass that otherwise fail, so re-added
32-bpo-39101.patch.3.9.1                already in 3.9.2
33-bpo-37961.patch.3.9.1                already in 3.9.2
34-bpo-41891.patch.3.9.1                already in 3.9.2
35-20142.patch.3.9.1                    old patch adapted to new code; correct?
41-omnios-fix-test_pkgutil.patch       let test_pkgutil fail. Thus, removed

So, actually this has one failing test less than our actual 3.9.1 version (on my systems)

alarcher commented:
@AndWac Thank you, I cannot remember exactly the details but I integrated some changes referenced in https://bugs.python.org/issue38323 that are not merged (or not completely).

alarcher commented:
@AndWac I have pushed the test results produced in my environment.
